content/guides/weird_characters.adoc

```diff
@@ -0,0 +1,846 @@
+
+[source,clojure]
+----
+user=> [1 2 3 #_ 4 5]
+[1 2 3 5]
+----
+Note that the space following `#_` is optional, so
+[source,clojure]
+----
+user=> [1 2 3 #_4 5]
+[1 2 3 5]
+----
+also works. Also note that the discard macro works for EDN
+
+The docs suggest that "The form following `#_` is completely skipped by the reader
+(This is a more complete removal than the `comments` macro which yields `nil`).".
+This can prove useful for debugging situations or for multiline comments.
+
+* <<xref/../../reference/reader#,Clojure Documentation - Reader>>
+* https://github.com/edn-format/edn#tagged-elements[EDN Tagged Elements]
+
+== `#"` - Regular Expression macro
+// " for the pleasure of emacs.
+
+See <<xref/../weird_characters#dispatch,(`#`)>> for additional details.
+
+`#"` indicates the start of a regular expression
+// "
+[source,clojure]
+----
+user=> (re-matches #"^test$" "test")
+"test"
+----
+
+This form is compiled at *read time* into a host-specific regex machinery, but
+it is not available in EDN.
+
+* <<xref/../../reference/other_functions#regex,Clojure Documentation: Regex Support>>
+
+== `#(` - Function macro
+
+See <<xref/../weird_characters#dispatch,(`#`)>> for additional details.
+
+`#(` begins the short hand syntax for an inline function definition. The
+following two snippets of code are similar:
+
+[source,clojure]
+----
+; anonymous function taking a single argument and printing it
+(fn [line] (println line))
+
+; anonymous function takin a single argument and printing it - shorthand
+#(println %)
+----
+
+The macro expands the shorthand syntax into a function definition whose
+arity (the number of arguments it takes) is defined by how the `%` placeholders
+are declared. See the `%` character for discussion around arity.
+
+[source,clojure]
+----
+user=> (macroexpand `#(println %))
+(fn* [arg] (clojure.core/println arg)) ; argument names shortened for clarity
+----
+
+== `#'` - Var macro
+
+`#'` is the var quote which expands into the `var` function:
+
+[source,clojure]
+----
+user=> (read-string "'foo")
+(var foo)
+user=> (def nine 9)
+#'user/nine
+user=> nine
+9
+user=> (var nine)
+#'user/nine
+user=> #'nine
+#'user/nine
+----
+When used it will attempt to return the referenced var. This is useful when
+you want to talk about the reference/declaration instead of the value it represents.
+See the use of `meta` int the metadata (<<xref/../weird_characters#metadata,`^`>>) discussion.
+
+Note that the var quote is not available in EDN.
+
+* <<xref/../../reference/special_forms#var,Clojure Official Documentation: Special Forms>>
+
+[[tagged_literals]]
+== `#inst`, `#uuid`, and `#js` etc. - tagged literals
+
+Commonly found in EDN and Clojure/ClojureScript this use of `#` is called
+the _tagged literal_. Look at this example:
+[source,clojure]
+----
+user=> (java.util.Date.)
+#inst "2014-05-19T19:12:37.925-00:00"
+----
+
+When we print a date it is represented as a tagged literal, or in this case,
```

Comment:
This sentence is confusing. "Tagged string" is not a thing. Printing is a generic operation that converts data to a string. Some data is printed in tagged literal format. But I think it's actually more confusing to include printing in this discussion instead of focusing on reading.